Count empty translations as ready for MT in the dashboard

seluianova opened the pull request:
### Short description
The value of "ready for MT" in the dashboard calculates the percentage of all pages with HIX above the threshold in relation to all translations.
This is incorrect since e.g. empty pages without a HIX score are allowed for MT


### Proposed changes
Add the number of empty translations to the ready_for_mt_count value


### Side effects
No?


### Resolved issues
Fixes: #2317
Fixes: #2296
Fixes: #2310 (self-resolved as a side effect)


__________________________________________________
<!-- Keep this link for the potential reviewer -->
[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

JoeyStk reviewed (approved):
Thank you a lot for the quick fix! :muscle: 

You could think about also including #2296 in this PR. In that issue we are asked to change the percentage (58% pages) to actual numbers (10 out of 17 pages). In combination with this PR the service-team asked us to write something like "10 out of 17 pages are ready for mt, 3 of them are empty pages and therefore not listed below" or the other way around "7 out of 17 pages are ready for mt, 3 more pages are empty and therefore not listed below". The service-team left it to us to decide which one we like better. If you think this would exceed the scope of this PR we can do it in a separate PR

timoludwig commented on integreat_cms/cms/views/dashboard/dashboard_view.py:
```diff
@@ -70,18 +70,25 @@ def get_hix_context(self):
 
+        # Get all translations with empty content
+        hix_translations_with_empty_content = [
+            pt for pt in hix_translations if not pt.content
```
```suggestion
            pt for pt in hix_translations if not pt.content.strip()
```
In the HIX signals, we also count a bunch of whitespaces as "empty"

timoludwig reviewed (changes requested):
Thanks! :+1:

seluianova commented:
Added changes for #2296.
Currently the widget looks like this:
![image](https://github.com/digitalfabrik/integreat-cms/assets/115008338/53d69839-31eb-4d42-801e-0bead0392b70)
I chose this sentence because in addition to empty pages, we may have pages with no HIX score, for example, if the TextLab API returned an error.
`For 16 pages the HIX value is not calculated.` - this number includes empty pages & non-empty pages without HIX (but with HIX enabled).
But let me know if that's not what's needed.

I also replaced the release notes because changes #2296 completely overwrote #2317.

timoludwig reviewed (changes requested):
> I chose this sentence because in addition to empty pages, we may have pages with no HIX score, for example, if the TextLab API returned an error. `For 16 pages the HIX value is not calculated.` - so this number includes empty pages & non-empty pages without HIX (but with HIX enabled). But let me know if that's not what's needed.

Hmm, yeah definitely a good question how to handle this case... :thinking: 
In my opinion, we don't need to tell the users that some values are missing... In the end, we just allow MTs for any page translation where we're in doubt (because the textlab api returned an error), so from the user's perspective, I think they should just count as valid...

Maybe we could add a management command to retrieve all pages where the HIX value could not be retrieved, so we can investigate the issue and find out why the textlab api calls are failing... but that's probably out of scope for this issue...

seluianova commented:
> from the user's perspective, I think they should just count as valid...

So if I take the message below, do you think the number of such pages should be included into YY?

`XX out of YY pages are ready for automatic translation. ZZ of them are empty and therefore not listed below.`

For example, we have:

1 page with HIX >= 15
1 page with HIX < 15
1 page with no HIX

The message will say:
"1 out of 3 pages are ready for automatic translation", but the table will contain 2 rows.

I'm worried that it will look like a bug because not obvious why "out of 3".

timoludwig commented:
> @timoludwig so, having the message below, do you think the number of such pages should be included into YY?
> 
> `XX out of YY pages are ready for automatic translation. ZZ of them are empty and therefore not listed below.`

In my opinion, the "ZZ of them are empty ..." is already too much information. I think a simple "XX out of YY pages are ready ..." is enough for the users to estimate whether they need to seriously rework their content or whether it's ok like this.

So my suggestion for the calculation would be:
YY = "page translations which have HIX enabled" (including empty pages or pages with HIX errors)
XX = YY - "page translations with HIX < 15"

So from the user's perspective, HIX errors don't matter because they don't prevent MT. HIX errors are just interesting for the development team to investigate the issue, but this information is not needed on the dashboard.

> For example, we have:
> 
> 1 page with HIX >= 15 1 page with HIX < 15 1 page with no HIX (but not empty)
> 
> The message will say: "1 out of 3 pages are ready for automatic translation", but the table will contain 2 rows.
> 
> I'm worried that it will look like a bug because not obvious why "out of 3".

I think this is a very rare edge case, in most cases the regions will have so many pages that they cannot realistically count the numbers of rows, so I doubt that they will be confused if the numbers don't exactly match.

> **UPD**. As one more option, probably we could just display such pages in the table? With the dash or N/A as HIX VALUE. In addition it will help to find such pages quickly, and update the value... I mean, a management command would help here, but if we display all the lines, the numbers will at least match.

In my opinion, this would just cause more confusion because they do not understand why these pages do not have a value... Since we treat such pages as "ready for MT", we could only show them with an artificial score of 20 or whatever... but I guess not listing them there is also fine.

> **UPD-2**. Or another option is to simply exclude them along with those who have HIX disabled.

I think this does not make sense, since HIX disabled means they cannot be automatically translated and HIX missing means they can.

seluianova commented:
Updated message to: 

`XX out of YY pages are ready for automatic translation.`

> YY = "page translations which have HIX enabled" (including empty pages or pages with HIX errors)
> XX = YY - "page translations with HIX < 15"

osmers commented:
So how would the numbers behave if all pages have a value above 15? They XX and YY should match, right?
Bcs I received the question yesterday from a municipality: Their number said 85% but the list did not show any pages they would have to work on. And I think that is very confusing. Then it is better that they get the information that ZZ pages don't have content and therefore no HIX but are translateable.

seluianova commented:
> So how would the numbers behave if all pages have a value above 15? They XX and YY should match, right?

@osmers right.

osmers commented:
> > So how would the numbers behave if all pages have a value above 15? They XX and YY should match, right?
> 
> @osmers right.

Sorry to ask again - but to me, that does not seem to be the case if YY includes empty pages. Unless we count XX as empty pages + pages with HIX > 15

I think my point is that is important that if all pages are ready for translation XX and YY have to match to avoid confusion

seluianova commented:
> Unless we count XX as empty pages + pages with HIX > 15

@osmers Yes, that's what we do. We count XX as all pages ready for MT: with HIX >= 15 or empty.

> I think my point is that is important that if all pages are ready for translation XX and YY have to match to avoid confusion

Yes, XX and YY will match, if all pages are ready for MT.

Examples:
If we have 3 pages, all have HIX >= 15 —> `3 out of 3 pages are ready for automatic translation.`
If we have 3 pages: 1 with HIX >= 15, 1 with HIX < 15, 1 empty —> `2 out of 3 pages are ready for automatic translation.`
If we have 3 pages, 1 empty, 2 with HIX < 15 —> `1 out of 3 pages are ready for automatic translation.`

osmers commented:
@seluianova perfect - thank you for clarifying :)

timoludwig reviewed (approved):
Awesome, thanks! :tada:
